shared/js/devtools/panel.es6.js

```diff
@@ -172,13 +171,58 @@ function appendCallStack (cell, stack) {
     }
 }
 
-function shouldShowRow (className) {
-    const filter = document.getElementById(`display-${className}`)
-    return !filter || filter.checked
+/**
+ * General panel configuration that can change during its lifetime.
+ *
+ * Values below are the defaults.
+ */
+const panelConfig = {
+    rowVisibility: {
+        blocked: true,
+        ignored: true,
+        ignoredFirstParty: true,
+        redirected: true,
+        cookieHTTP: true,
+        cookieJS: true,
+        apiCanvas: true,
+        noneRequest: true
+    },
+    rowFilter: ''
+}
+
+function shouldShowRow (row) {
+    // empty search box is considered to be no filter
+    if (panelConfig.rowFilter !== '') {
+        // when a filter is in effect, fail now if the URL does not match the filter
+        if (!row.cells[1].textContent.match(panelConfig.rowFilter)) {
+            return false
+        }
+    }
+
+    const className = row.classList[0]
+    switch (className) {
+    case 'ignore':
+        if (row.querySelector('.request-action').textContent === `${actionIcons.ignore} ignore (first party)`) {
+            return panelConfig.rowVisibility.ignored && panelConfig.rowVisibility.ignoredFirstParty
+        }
+        return panelConfig.rowVisibility.ignored
+    case 'block':
+        return panelConfig.rowVisibility.blocked
+    case 'redirect':
+        return panelConfig.rowVisibility.redirected
+    case 'cookie-tracker':
+        return panelConfig.rowVisibility.cookieHTTP
```

Comment:
You should include set cookies here.
```suggestion
    case 'cookie-tracker':
    case 'set-cookie-tracker':
        return panelConfig.rowVisibility.cookieHTTP
```